gen_core_pools.py

```diff
@@ -0,0 +1,77 @@
+import json
+
+import requests
+
+from gen_pools_and_gauges import get_subgraph_url
+
+
+def get_stable_pools_with_rate_provider():
+    """
+    for every chain, query the official balancer subgraph and retrieve pools that:
+    - have a rate provider different from address(0)
+    - have a liquidity greater than $250k
+    - have a yield fee > 0
+    """
+    core_pools = {}
+    with open("extras/chains.json", "r") as f:
+        chains = json.load(f)
+    for chain in chains["CHAIN_IDS_BY_NAME"]:
+        if chain in ["sepolia", "goerli"]:
+            continue
+        core_pools[chain] = {}
+        url = get_subgraph_url(chain)
+        query = """{
```

Comment:
At some point amount of pools can be over 1000 across all chains. Maybe worth doing some pagination with offset? Not insisting as it might not be as important now

Reply:
yes, on some chains the page limit of 1000 is reached.

but that is only if querying the full dataset. here the fact that there are added conditions (have rate provider, liq>$250k, have yield fee) should decrease the size of that dataset substantially.